src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php

```diff
@@ -91,22 +99,62 @@ public function isRisky()
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
-        $functionNames = $this->getFunctionNames();
+        $functionNames = $this->getAllInternalFunctionsNormalized();
 
-        $indexes = [];
+        if ('namespaced' === $this->configuration['scope']) {
+            foreach (array_reverse($this->getUserDefinedNamespaces($tokens)) as $namespace) {
+                $this->fixFunctionCalls($tokens, $functionNames, $namespace['open'], $namespace['close']);
+            }
+        } else {
+            $this->fixFunctionCalls($tokens, $functionNames, 0, count($tokens) - 1);
+        }
+    }
 
-        for ($index = 0, $count = $tokens->count(); $index < $count; ++$index) {
-            $token = $tokens[$index];
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('exclude', 'List of functions to ignore.'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([function ($value) {
+                    foreach ($value as $functionName) {
+                        if (!\is_string($functionName) || '' === \trim($functionName) || \trim($functionName) !== $functionName) {
+                            throw new InvalidOptionsException(\sprintf(
+                                'Each element must be a non-empty, trimmed string, got "%s" instead.',
+                                \is_object($functionName) ? \get_class($functionName) : \gettype($functionName)
+                            ));
+                        }
+                    }
 
-            $tokenContent = $token->getContent();
+                    return true;
+                }])
+                ->setDefault([])
+                ->getOption(),
+            (new FixerOptionBuilder('scope', 'Fix functions only if called in given scope, global or within user defined namespaces only.'))
```

Comment:
>`global` or `within user-defined-namespaces` only.

would that help? Besides, maybe it is best to align on concept before details as it still unclear if this PR should be finished in the first place , sadly :(

Reply:
What confuses me is the _"user-defined namespaces"_ part. Maybe a boolean option like `ignore_global_space` would fit better? It could be described as _"Whether functions called from global space should be updated (defaults to `true`)."_

Anyway, I'm 👍 for this new configuration.

I would even go further, like config that allows to _remove_ (not just _ignore_) the backslashes. If accepted, this should probably be done in a separate PR but I wonder if it would be compatible with the option introduced here.